### ISSUE TITLE
Memory doubled after the first round

### ISSUE THREAD

**Issue by jonbakerfish** [facebook/fb.resnet.torch#84]
I'm trying [fine-tuning the resnet model](https://github.com/facebook/fb.resnet.torch/blob/master/pretrained/README.md#fine-tuning-on-a-custom-dataset). However, the GPU memory consumption doubled after the first epoch. What is the potential reason for such behavior?

I slightly modified the training scripts by freezing all the conv layers & batch-normalization layers:

    model:apply(function(m) if torch.type(m):find("BatchNormalization") then m:evaluate() end end)

    local layers = self.model:findModules('cudnn.SpatialConvolution')
    for i, m in ipairs(layers) do
      m.accGradParameters = function() end
      m.updateParameters = function() end
    end

**Comment by colesbury** [comment]
Make sure you're up to date (`git pull`). I fixed an issue with `-shareGradInput`/GPU memory after the first epoch yesterday.

I think the `m:evaluate()` call may not an effect. We call `model:training()` at the start of the [training loop](https://github.com/facebook/fb.resnet.torch/blob/master/train.lua#L49), which will override it.

**Comment by wuyuebupt** [comment]
Thanks for the update. Is there anyway to call no additional gpu memory compared with the first epoch?

**Comment by jonbakerfish** [comment]
@colesbury It works. Thank!
